Nos/Router.swift

```diff
@@ -152,6 +171,13 @@ extension Router {
                     }
                 } else if link.hasPrefix("%") {
                     pushNote(id: identifier)
+                } else if link.hasPrefix("$") {
+                    let parts = identifier.split(separator: ",").map { String($0) }
```

Comment:
Oh yeah, you're 100% right. I guess I assumed `d` tags couldn't contain a comma, but I don't see anything about that. I'm not actually sure what the rules are for `d` tags, so I suppose I can just use some more obscure character or three...

Reply:
Yeah, so I tried this and it looks to me like you can indeed have a `d` tag with a comma:

`% nak encode naddr -d abc,123 -p 0267aa3d92d2a479ad6bccdc6fe7657037deab4b77a8bbcfd3663b0eef196b58 -k 30023`

I wasn't able to figure out how to test this since all branches just call Router itself. I could subclass Router and override a `push()` method and then spy on what's happening but that feels icky. I dunno; I wish this `open(url:with:)` method were outside of Router, in another type, and called into Router. That would make it easier to test. Maybe I could even do that refactor here.

At any rate, I changed it from a comma to a semicolon which is hopefully less of an issue.

Reply:
Won't a semicolon have the same problem as a comma? As far as I understand the d tag can be an arbitrary string, so any delimiter we use could appear inside the tag. But what we could do is instead of assuming the delimiter only appears once in `link` and using `parts.first` and `parts.last` we could assume the delimiter appears multiple times, fetch the author ID using `parts.last`, then take `parts.removingLast()` and recombine them to get the dTag.

Also I can imagine an integration test without subclassing Router because `push` mutates one of the `NavigationPath`s which are published publicly. So something like:
```
let subject = Router()
let expectedEvent = try Event.findOrCreateBy(replaceableID: "498349827", authorID: "981247891273981723")
subject.open(url: URL(string: "$498349827;981247891273981723"))
XCTAssertEqual(subject.homeFeedPath.last, .note(.replaceableIdentifier(replaceableID: "981247891273981723", author: "981247891273981723")))
```

Reply:
Cool, I can try those both. Thanks!

Reply:
As for getting the _last_ of the parts and recombining the rest: no problem, done.

As for the test: I don't see a way to get the `.last` of the `homeFeedPath`, which is a `NavigationPath`. I'm not finding anything in the [docs](https://developer.apple.com/documentation/swiftui/navigationpath) or in experimentation or the internet. Oh, and when I print `subject.homeFeedPath`, I get this:

`NavigationPath(_items: SwiftUI.NavigationPath.(unknown context at $1cd8e6eec).Representation.eager([]), subsequentItems: [], iterationIndex: 0)`

so that's not an option, either. Thoughts?

Reply:
Oh yeah, I think I have run into this before. `NavigationStack` will actually take as a `path` pretty much any collection. So we could refactor our code to an array of `NosNavigationDestination` as our path instead of `NavigationPath`.